### PR TITLE
Replace redirects - C# and VB

### PR DESCRIPTION
**Pull request by gewarren**

### CODE REVIEW

File: docs/csharp/language-reference/compiler-options/target-compiler-option.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ The **-target** compiler option can be specified in one of four forms:
  [-target:winmdobj](./target-winmdobj-compiler-option.md)  
  To create an intermediate .winmdobj file.  
   
- Unless you specify **-target:module**, **-target** causes a .NET Framework assembly manifest to be placed in an output file. For more information, see [Assemblies in .NET](../../../standard/assembly/index.md) and [Common Attributes](../../programming-guide/concepts/attributes/common-attributes.md).  
+ Unless you specify **-target:module**, **-target** causes a .NET Framework assembly manifest to be placed in an output file. For more information, see [Assemblies in .NET](../../../standard/assembly/index.md) and [Common Attributes](../attributes/global.md).
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-options/target-compiler-option.md:34]: Thanks for catching this. I thought I had found all these links when I updated the related articles.

File: docs/csharp/misc/cs0612.md [hunk 2]
```diff
@@ -7,10 +7,11 @@ helpviewer_keywords:
   - "CS0612"
 ms.assetid: 7695f3b7-ffef-43f7-83db-fc1a9e361f1a
 ---
-# Compiler Warning (level 1) CS0612
+# Compiler warning (level 1) CS0612
+
 'member' is obsolete  
   
- The class designer marked a member with the [Obsolete](../programming-guide/concepts/attributes/common-attributes.md#Obsolete) attribute. This means that the member might not be supported in a future version of the class.  
+ The class designer marked a member with the <xref:System.ObsoleteAttribute> attribute. This means that the member might not be supported in a future version of the class.
```
Comment (BillWagner) [inline docs/csharp/misc/cs0612.md:14]: This is now targeting a different article, but I think that's fine.

File: docs/csharp/whats-new/version-update-considerations.md [hunk 3]
```diff
@@ -57,4 +57,4 @@ New code:
 public double CalculateSquare(in double value) => value * value;
 ```
 
-The [What's new](index.md) articles note if introducing a feature that affects public declarations is source compatible or binary compatible.
+The [What's new](csharp-8.md) articles note if introducing a feature that affects public declarations is source compatible or binary compatible.
```
Comment (BillWagner) [inline docs/csharp/whats-new/version-update-considerations.md:60]: This one should stay as is.  The `index.md` file will always be redirected to the latest version (when I add csharp-9.md, for example). That way there's one location to update.

```suggestion
The [What's new](index.md) articles note if introducing a feature that affects public declarations is source compatible or binary compatible.
```

**Review by BillWagner — APPROVED** [review]
Overall this is great @gewarren 

I had one comment. Is there a way to tag those redirects that are by design?

File: docs/csharp/whats-new/version-update-considerations.md (hunk 3 above)
Comment (gewarren) [reply docs/csharp/whats-new/version-update-considerations.md:60]: Ah, makes sense. I should check for this in my other PRs. Currently my tool doesn't have a way of excluding redirects, but I'll know to look for this in the future (there's always some manual manipulation involved, e.g. to fix link text).

File: docs/csharp/misc/cs0612.md (hunk 2 above)
Comment (gewarren) [reply docs/csharp/misc/cs0612.md:14]: I couldn't find a mention of Obsolete in common-attributes.md. Let me know if I just missed it.

Comment (BillWagner) [reply docs/csharp/misc/cs0612.md:14]: It's here: https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/attributes/general.md#obsolete-attribute

**Comment by gewarren** [comment]
@BillWagner Could you merge this one? The WIP check is acting up.

**Pull request merged**